mods/stairs/init.lua

```diff
@@ -4,8 +4,8 @@
 stairs = {}
 
 -- Node will be called stairs:stair_<subname>
-function stairs.register_stair(subname, recipeitem, groups, images, description, sounds)
-	minetest.register_node(":stairs:stair_" .. subname, {
+function stairs.register_stair(subname, recipeitem, groups, images, description, sounds, def)
```

Comment:
instead of adding def as an additional parameter to the function, you could use the third one (groups) as def if images isn't given. Then you could define the groups, tiles (images), description, sounds and more in the third parameter (def (or groups))